### PR TITLE
Merge-1. (Partial)

### PR DESCRIPTION
**Pull request by pbhide**
Work in progress

### CODE REVIEW

File: targets/switch/README.md [hunk 1]
```diff
@@ -67,9 +67,9 @@ By default, DOCKER_IMAGE is set to 'bm-switchlink'.
 Invoking make without an explicit target builds the softswitch with only the
 auto-generated API (make bm).
 
-For details on the features supported by switchapi, switchsai, switchlink, and p4ofagent
-libraries, please refer to the README.md file in the switchapi, switchsai, switchlink,
-and p4ofagent repositories, respectively.
+For details on the features supported by switchapi, switchsai, and switchlink
```
Comment (cruzj) [inline targets/switch/README.md]: Looks like you are removing p4ofagent from the README file. Is this intentional?

File: targets/switch/p4src/openflow.p4 [hunk 2]
```diff
@@ -80,7 +96,7 @@ action nop () {
 }
 
 action terminate_cpu_packet() {
-    modify_field(standard_metadata.egress_spec,fabric_header.dstPortOrGroup);
+    modify_field(standard_metadata.egress_port,fabric_header.dstPortOrGroup);
```
Comment (cruzj) [inline targets/switch/p4src/openflow.p4]: I think it is egress_spec and not egress_port.

Comment (pierce-m) [inline targets/switch/p4src/openflow.p4]: Egress spec is valid for ingress pipeline while egress port is valid for egress pipeline in p4lang. So this should remain "egress spec."

**Comment by pbhide** [comment]
Updated. Ready for review.

File: targets/switch/of-tests/tests/api-tests/switch_acl.py [hunk 3]
```diff
@@ -241,35 +248,47 @@ class MirrorSessionTest(api_base_tests.ThriftInterfaceDataPlane):
...
+        self.client.switcht_api_mirror_session_delete(0, mirror1)
+        self.client.switcht_api_mirror_session_delete(0, mirror2)
+        self.client.switcht_api_mirror_session_delete(0, mirror3)
         # delete again -ve test
-        self.client.switcht_mirror_session_delete(0, 201)
+        self.client.switcht_api_mirror_session_delete(0, mirror3)
 
 class MirrorAclTest_e2e(api_base_tests.ThriftInterfaceDataPlane):
     def runTest(self):
-        if is_bmv2:
```
Comment (antoninbas) [inline targets/switch/of-tests/tests/api-tests/switch_acl.py]: Please do not remove this code...

File: targets/switch/of-tests/tests/api-tests/switch_acl.py [hunk 4]
```diff
@@ -357,23 +386,20 @@ def runTest(self):
 
 class MirrorAclTest_i2e_erspan(api_base_tests.ThriftInterfaceDataPlane):
     def runTest(self):
-        if is_bmv2:
```
Comment (antoninbas) [inline targets/switch/of-tests/tests/api-tests/switch_acl.py]: same as above, do not remove

**Comment by antoninbas** [comment]
Could we put new tests (related to INT I presume) in a separate Python file instead of stuffing everything in the same file (switch.py)?

**Comment by pbhide** [comment]
All INT tests are in switch_int.py.
Which ones are you refering to? Mirroring tests? They are in switch_acl.py.

**Comment by antoninbas** [comment]
I saw several tests additions to switch.py (the file is ~2000 lines longer). What are these and what are they doing in this PR if they are not INT related?
Also, please remember to disable the tests which fail with bmv2 like this: https://github.com/p4lang/p4factory/blob/master/targets/switch/of-tests/tests/api-tests/switch_acl.py#L139 and send an email with the list of tests. If it's more than mirroring tests, there is a problem...
Note that the travis regression does not run the switchapi tests with bmv2 (testing is limited to 50 minutes), which is why it is important to run then on your machine.

**Pull request merged**